### PR TITLE
Connection Management for SFMC Source and Sink.

### PR DESCRIPTION
**Pull request by Vipinofficial11**

### CODE REVIEW

**Comment by Vipinofficial11** [comment]
@rmstar please review.

File: src/main/java/io/cdap/plugin/sfmc/common/SalesforceMarketingCloudBaseConfig.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/sfmc/common/SalesforceMarketingCloudBaseConfig.java]: 2022

File: src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) {
+    return null;
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java]: implement this

File: src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  @Override
+  public ConnectorSpec generateSpec(ConnectorContext connectorContext, ConnectorSpecRequest connectorSpecRequest) {
+    return null;
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java]: implement this

File: src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import io.cdap.plugin.sfmc.source.util.MarketingCloudConstants;
+
+/**
+ * Salesforce Marketing Cloud Connector
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(MarketingCloudConstants.PLUGIN_NAME)
+@Description("Connection to access data from Salesforce Marketing Objects.")
+
+public class MarketingConnector implements Connector {
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/sfmc/connector/MarketingConnector.java]: add support for sampling

**Review by rmstar — COMMENTED** [review]
Add @Metadata annotation, see other connection PRs

**Comment by vikasrathee-cs** [comment]
Closing this PR due to CLA issue. WIll raise a new one.

**Pull request closed without merging**